sdk/src/process_instruction.rs

```diff
@@ -314,7 +373,25 @@ impl InvokeContext for MockInvokeContext {
         Ok(())
     }
     fn get_caller(&self) -> Result<&Pubkey, InstructionError> {
-        Ok(&self.key)
+        self.invoke_stack
+            .last()
+            .map(|frame| &frame.key)
+            .ok_or(InstructionError::GenericError)
+    }
+    fn remove_first_keyed_account(&mut self) -> Result<(), InstructionError> {
+        let stack_frame = &mut self
+            .invoke_stack
+            .last_mut()
+            .ok_or(InstructionError::GenericError)?;
+        stack_frame.keyed_accounts_range.start =
```

Comment:
Doing this to avoid re-arranging the keyed accounts list?

Reply:
It avoids possibly reallocating the `Vec` if we were to call `pop` or `remove` instead. And for the `KeyedAccount`s to be mapped directly it is beneficial to have them stay at one constant pointer over their lifetime.

Reply:
Yeah, that's what I figured, be nice to clean up this later